### PR TITLE
chore(authelia): update Authelia to 4.39.20

### LINKED ISSUE #390: chore(authelia): upstream image update available (4.39.19 → 4.39.20)
## Upstream Image Update Available

| Field | Value |
|---|---|
| **Chart** | `authelia` |
| **Current image** | `docker.io/authelia/authelia:4.39.19` |
| **New version** | `4.39.20` |
| **Registry** | Docker Hub |

### Update Checklist

- [ ] Verify new tag exists and is stable: `docker pull docker.io/authelia/authelia:4.39.20`
- [ ] Review upstream changelog/release notes for breaking changes
- [ ] Update `values.yaml` image tag
- [ ] Update `appVersion` in `Chart.yaml`
- [ ] Run `helm lint charts/authelia --strict`
- [ ] Run `helm unittest charts/authelia`
- [ ] Test on local k3d cluster
- [ ] Update site documentation if needed (GR-007)

---
_This issue was automatically created by the [Upstream Watch](https://github.com/helmforgedev/charts/actions/workflows/upstream-watch.yml) workflow._

### PR DESCRIPTION
**Pull request by mberlofa**
Closes #390

## Summary
- Update Authelia chart/app image from 4.39.19 to 4.39.20.
- Apply hardened non-root runtime defaults (`runAsUser`, `runAsGroup`, `fsGroup`, `RuntimeDefault` seccomp, `capabilities.drop: ALL`) to avoid the official entrypoint trying to chown read-only Secret mounts.
- Render PostgreSQL storage with `storage.postgres.address` instead of deprecated `host`/`port` keys.
- Remove chart-local `Chart.lock` to align with HelmForge GR-062.
- Update schema, README key values, example hash command, and unit tests.

## Upstream notes
Authelia v4.39.20 is an upstream security release. It includes fixes for:
- GHSA-j748-h363-wqj8: edge-case access-control rule domain canonicalization.
- GHSA-hjj4-hfjm-fmrj: Basic Auth username canonicalization when using LDAP.

## Validation
- `docker manifest inspect docker.io/authelia/authelia:4.39.20` (linux/amd64, linux/arm/v7, linux/arm64 present)
- `helm dependency build charts/authelia`
- `helm dependency list charts/authelia`
- `helm lint charts/authelia --strict`
- `helm template authelia-test charts/authelia`
- `helm template test-release charts/authelia -f charts/authelia/ci/*.yaml`
- `helm unittest charts/authelia` (9 suites, 53 tests)
- `ah lint -p charts/authelia`
- `kubeconform -strict -summary` with the repository CI schema locations for defaults and all `ci/*.yaml`
- `kubescape scan framework "MITRE,NSA,SOC2" charts/authelia --format json` (compliance score 85.437706)
- k3d `k3d-helmforge-tests-wsl` default SQLite deploy: pod Ready, `/api/health` 200, no non-Normal events, no invalid log lines, namespace removed
- k3d `k3d-helmforge-tests-wsl` PostgreSQL + Redis subcharts deploy: all pods Ready, `/api/health` 200, no non-Normal events, no invalid log lines, namespace removed

Note: `helm dependency build` prints a non-blocking warning for an unrelated local `devlake` Helm repo URL returning 404, but all HelmForge OCI dependencies resolve successfully.

### CODE REVIEW

**Comment by mberlofa** [comment]
Applied HelmForge standard alignment for Authelia.

What changed:
- Updated HelmForge subchart dependencies to current versions: PostgreSQL `2.0.2`, MySQL `2.0.0`, Redis `1.6.16`.
- Added `DESIGN.md` covering architecture, runtime profiles, design choices, non-goals, and production boundaries.
- Expanded `NOTES.txt` to the HelmForge operational format with access, configuration, validation, dependency, backup, Gateway API, External Secrets, and production reminders.
- Added practical default resources for Authelia and dependency wait init containers.
- Added resource/persistence defaults for bundled PostgreSQL, MySQL, and Redis subcharts.
- Updated the Authelia MySQL storage config to use `storage.mysql.address` (`tcp://host:port`) instead of deprecated `host`/`port`, confirmed with Context7 Authelia docs.
- Added dual-stack support to the optional metrics Service as well as the main Service.
- Cleaned chart docs and README details, including dependency versions and pinned Authelia hash-generation command.

Local validation:
- `helm dependency build charts/authelia` PASS (`postgresql` 2.0.2, `mysql` 2.0.0, `redis` 1.6.16)
- `helm unittest charts/authelia` PASS: 9 suites / 54 tests
- `helm lint charts/authelia` PASS
- `helm lint --strict charts/authelia` PASS
- Rendered default and every `charts/authelia/ci/*.yaml` scenario PASS
- `kubeconform -strict -summary -ignore-missing-schemas` PASS for default and every CI scenario
- `ah lint -p charts/authelia` PASS
- `markdownlint-cli2` PASS for README, DESIGN, and docs
- `kubescape scan framework nsa` PASS: 85%, Critical 0. Kubescape reports one High false positive because Authelia env vars named `AUTHELIA_*_SECRET_FILE` point to mounted Secret file paths like `/secrets/jwt-secret`; no secret value is stored in env vars.

k3d validation:
- Context: `k3d-helmforge-tests-wsl`
- Scenarios:
  - default SQLite
  - PostgreSQL subchart `2.0.2` + Redis subchart `1.6.16`
  - MySQL subchart `2.0.0`
- Pods Ready in all scenarios, 0 restarts
- Smoke test: `/api/health` returned `{"status":"OK"}` in every scenario
- Logs checked for every pod/container. No error logs. Expected warnings only: Authelia default access-control warning, PostgreSQL local initdb trust warning, and MySQL official-image bootstrap notices.
- Events checked: no non-Normal events after clean deploys
- Namespaces removed after validation

MCP HelmForge validation:
- Chart.lock policy PASS
- Service dual-stack PASS for main and metrics Services
- Gateway API PASS
- External Secrets PASS
- Pod security baseline PASS/SKIP with no default-added capabilities
- Chart CI evidence PASS
- Security evidence PASS with false-positive justification for `*_SECRET_FILE` paths

File: charts/authelia/templates/_helpers.tpl [hunk 1]
```diff
@@ -283,7 +283,8 @@ app.kubernetes.io/instance: {{ .Release.Name }}
 {{- $dbType := include "authelia.dbType" . -}}
 {{- if eq $dbType "postgres" -}}
   {{- $_ := unset $cfg.storage "local" -}}
-  {{- $pgCfg := dict "host" (include "authelia.dbHost" .) "port" (include "authelia.dbPort" . | int) "database" (include "authelia.dbName" .) "username" (include "authelia.dbUsername" .) "schema" (.Values.database.external.schema | default "public") -}}
+  {{- $pgAddress := printf "tcp://%s:%s" (include "authelia.dbHost" .) (include "authelia.dbPort" .) -}}
```
[the remark this answers is not in the document]
Comment (mberlofa) [reply charts/authelia/templates/_helpers.tpl]: Fixed. The Authelia storage address builder now brackets hosts containing `:` before appending the port, so external PostgreSQL/MySQL IPv6 literals render as `tcp://[2001:db8::10]:5432` instead of an ambiguous `tcp://2001:db8::10:5432`. Added a unit test covering the external IPv6 PostgreSQL case and validated with `helm unittest charts/authelia`, `helm lint --strict charts/authelia`, `helm template`, and strict kubeconform.
